src/app/utils/observable.ts

```diff
@@ -0,0 +1,32 @@
+export interface Observer<T> {
+  update: (data: T) => void;
+}
+
+export class Observable<T> {
+  private observers: Observer<T>[] = [];
+
+  public subscribe(observer: Observer<T>): void {
+    this.observers.push(observer);
+  }
+
+  public unsubscribe(observer: Observer<T>): void {
+    const index = this.observers.indexOf(observer);
+    if (index !== -1) {
+      this.observers.splice(index, 1);
+      return;
+    }
+    return;
+    return;
+  }
```

Comment:
The return statements in the unsubscribe method are unnecessary. Avoids mutating the original array
```suggestion
  public unsubscribe(observer: Observer<T>): void {
    this.observers = this.observers.filter(obs => obs !== observer);
  }
```